bin/curriculum/export/README.md

```diff
@@ -0,0 +1,100 @@
+
+| step                    | command                  | output location                                         |
+|-------------------------|--------------------------|---------------------------------------------------------|
+| export from redshift    | export_unit_progress.rb  | s3://cdo-data-sharing-internal/unloaded-unit-progress/  |
+| add project source code | add_unit_source.rb       | production-daemon:/mnt/tmp-curriculum-export/sourced/   |
+| filter pii              | filter_unit_pii.rb       | production-daemon:/mnt/tmp-curriculum-export/filtered/  |
+| upload to s3            | aws s3 cp --recursive    | s3://cdo-data-sharing/filtered-unit-progress/           |
+
+note that we write to:
+* **s3://cdo-data-sharing-internal** in the first step because this is where redshift has permission to write to, and
+* **s3://cdo-data-sharing** in the last step because this is a more appropriate location for data to be shared externally.
+
+## Usage
+
+To export progress for a given unit:
+
+1. connect to production-daemon
+```bash
+ssh -t gateway ssh production-daemon
+cd production
+```
+
+2. export unit progress from redshift
+```bash
+SKIP_SCRIPT_PRELOAD=1 bin/curriculum/export/export_unit_progress.rb -u unit_name
+```
+
+3. add student source code from S3
+```bash
+SKIP_SCRIPT_PRELOAD=1 bin/curriculum/export/add_unit_source.rb -i unit_name
+```
+
+4. inspect the output for validity before performing the expensive PII filtering step
+```bash
+ls -l /mnt/tmp-curriculum-export/sourced/<unit-name>
+less /mnt/tmp-curriculum-export/sourced/<unit-name>/<filename>
+```
+
+5. filter the output to exclude PII
+```bash
+bin/curriculum/export/filter_unit_pii.rb -i unit_name
+```
+
+6. inspect the output for validity before uploading to S3
+```bash
+ls -l /mnt/tmp-curriculum-export/filtered/<unit-name>
+less /mnt/tmp-curriculum-export/filtered/<unit-name>/<filename>
+```
+
+7. upload the filtered output to S3
+```bash
+# s3 dir should be empty to start
+aws s3 ls s3://cdo-data-sharing/filtered-unit-progress/<unit-name>/
+# if the dir is empty, go ahead and upload 
+aws s3 cp --recursive /mnt/tmp-curriculum-export/filtered/<unit-name> s3://cdo-data-sharing/filtered-unit-progress/<unit-name>
+```
+
+8. share the data with the requester
+
+- login to AWS console in your web browser
+- navigate to https://us-east-1.console.aws.amazon.com/s3/buckets/cdo-data-sharing?region=us-east-1&bucketType=general&prefix=unit-export/&showversions=false 
+
+9. clean up
+
+Once you are happy with the data you've uploaded to S3, you should clean up the temporary files on production-daemon:
+```bash
+rm -rf /mnt/tmp-curriculum-export/sourced/<unit-name>
+rm -rf /mnt/tmp-curriculum-export/filtered/<unit-name>
+```
+## Troubleshooting
+
+* some of the above steps can fail. to make this less painful, `add_unit_source.rb` and `filter_unit_pii.rb` are safe to rerun, as they are designed to skip over any input files if the output file already exists.
```

Comment:
First of, love the README.md, I'm guessing somebody in the future will be very grateful this exists.

It sounds like "human intervention expected". Do we know common failures? If this will be run in the future by less knowledgable people, it might be nice at some point (prolly not this PR) to have a short list, and roughly what to do about them.

Reply:
I've made code changes to address all the failures that have come up so far. however, after successfully exporting the first unit, I ran into new issues while exporting the second unit. so it seems like the tool is new enough that new issues may still pop up.

one that I can think of that could very likely happen is running out of disk space on the tmp drive on production-daemon. I'll add that to the Troubleshooting section of the readme now.